lib/cryptoexchange/exchanges/dydx_perpetual/services/pairs.rb

```diff
@@ -0,0 +1,27 @@
+module Cryptoexchange::Exchanges
+  module DydxPerpetual
+    module Services
+      class Pairs < Cryptoexchange::Services::Pairs
+        PAIRS_URL = "#{Cryptoexchange::Exchanges::DydxPerpetual::Market::API_URL}/stats/markets"
+
+        def fetch
+          output = super
+          adapt(output)
+        end
+
+        def adapt(output)
+          output['markets'].map do |pair, value|
+            next if value["type"] != "PERPETUAL"
```

Comment:
verify that all tickers from `PAIRS_URL` is futures tickers then remove this line and fetch all tickers from `PAIRS_URL`